frontend/tests/controllers/CertificatesTest.php

```diff
@@ -56,12 +55,149 @@ public function testGenerateContestCertificates() {
+        $timeFuture =  new \OmegaUp\Timestamp($currentTime + 60 * 60);
+        $contestData = \OmegaUp\Test\Factories\Contest::createContest(
+            new \OmegaUp\Test\Factories\ContestParams([
+                'admissionMode' => 'public',
+                'startTime' => $timePast,
+                'finishTime' => $timeFuture,
+            ])
+        );
+
+        \OmegaUp\Test\Factories\Contest::addAdminUser(
+            $contestData,
+            $certificateGenerator
+        );
+
+        //Create and add problems to the contest
+        $problems = [];
+        for ($i = 0; $i < 5; $i++) {
+            $problems[$i] = \OmegaUp\Test\Factories\Problem::createProblem();
+
+            \OmegaUp\Test\Factories\Contest::addProblemToContest(
+                $problems[$i],
+                $contestData
+            );
+        }
+
+        //Associate the identity index with
+        //the problem indexes they are going to send
+        $submissions = [
+            0 => ['problems' => [0, 1, 2, 3, 4]],
+            1 => ['problems' => [1]],
+            2 => ['problems' => [0, 1, 2, 3]],
+            3 => ['problems' => [2]],
+            4 => ['problems' => [3]],
+            5 => ['problems' => [0]],
+            6 => ['problems' => [0, 1]],
+            7 => ['problems' => [0, 1, 2]],
+        ];
+
+        //Send submissions
+        foreach ($submissions as $identityIndex => $problemsOfIdentity) {
+            foreach ($problemsOfIdentity['problems'] as $problemIndex) {
+                $run = \OmegaUp\Test\Factories\Run::createRun(
+                    $problems[$problemIndex],
+                    $contestData,
+                    $identities[$identityIndex]
+                );
+                \OmegaUp\Test\Factories\Run::gradeRun($run, 1.0, 'AC', 10);
+            }
+        }
+
+        //Finish the contest to generate certificates
+        $contestData['contest']->finish_time = \OmegaUp\Time::get() - 1;
+        \OmegaUp\DAO\Contests::update($contestData['contest']);
+
+        $certificatesCutoff = 4;
+
+        $response = \OmegaUp\Controllers\Certificate::apiGenerateContestCertificates(
+            new \OmegaUp\Request([
+                'auth_token' => $loginIdentity->auth_token,
+                'contest_id' => $contestData['contest']->contest_id,
+                'certificates_cutoff' => $certificatesCutoff
+            ])
+        );
+
+        //Assert status of new contest
+        $this->assertSame('ok', $response['status']);
+
+        $contest = \OmegaUp\DAO\Contests::getByAlias(
+            $contestData['request']['alias']
+        );
+        $this->assertEquals(4, $contest->certificate_cutoff);
+
+        $certificates = \OmegaUp\DAO\Certificates::getAll();
+        $this->assertEquals(0, sizeof($certificates));
+
+        \OmegaUp\Test\Utils::runGenerateContestCertificates();
+
+        $certificates = \OmegaUp\DAO\Certificates::getAll();
+        $this->assertEquals(8, sizeof($certificates));
```

Comment:
Nit: 
```suggestion
        $this->assertCount(8, $certificates);
```
Hay una función en [PHPUnit ](https://docs.phpunit.de/en/9.6/assertions.html#assertcount)para contar elementos de array, así que y ano hay necesidad de calcular el tamaño con `sizeof`